File: FileHandling/app/src/main/java/com/example/filehandling/MainActivity.java
package com.example.filehandling;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import java.io.FileInputStream;
import java.io.FileOutputStream;

public class MainActivity extends AppCompatActivity {

    Button b1,b2;
    TextView tv;
    EditText et1;
    String file="mydata.txt";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        b1=findViewById(R.id.b1);
        b2=findViewById(R.id.b2);
        tv=findViewById(R.id.tv);
        et1=findViewById(R.id.et1);

        b1.setOnClickListener(v->{
            String data=et1.getText().toString();
            try{
                FileOutputStream fos=openFileOutput(file,MODE_APPEND);
                fos.write(data.getBytes());
                fos.close();

                Toast.makeText(this, "Data Saved", Toast.LENGTH_SHORT).show();
        }catch (Exception e){
                e.printStackTrace();
            }
            });

        b2.setOnClickListener(v->{
                try{
                    FileInputStream fin=openFileInput(file);
                    int c;
                    String data="";
                    while ((c=fin.read())!=-1){
                        data=data+(char)c;
                    }
                    tv.setText(data);
                    fin.close();

                    Toast.makeText(this, "Data Retrieved", Toast.LENGTH_SHORT).show();
                }catch (Exception e){
                    e.printStackTrace();
                }

        });
    }
}

File: Second/app/src/main/java/com/example/second/MainActivity.kt
package com.example.second

import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        Log.d("lifecycle", "onCreate invoked")
        Toast.makeText(this, "onCreate Called", Toast.LENGTH_LONG).show()

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }

    override fun onStart() {
        super.onStart()
        Log.d("lifecycle", "onStart invoked")
        Toast.makeText(this, "onStart Called", Toast.LENGTH_LONG).show()
    }

    override fun onResume() {
        super.onResume()
        Log.d("lifecycle", "onResume invoked")
        Toast.makeText(this, "onResume Called", Toast.LENGTH_LONG).show()
    }

    override fun onPause() {
        super.onPause()
        Log.d("lifecycle", "onPause invoked")
        Toast.makeText(this, "onPause Called", Toast.LENGTH_LONG).show()
    }

    override fun onStop() {
        super.onStop()
        Log.d("lifecycle", "onStop invoked")
        Toast.makeText(this, "onStop Called", Toast.LENGTH_LONG).show()
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d("lifecycle", "onDestroy invoked")
        Toast.makeText(this, "onDestroy Called", Toast.LENGTH_LONG).show()
    }
}
